File: CSharp/Controls/LogViewer.Wpf/Extensions/ServicesExtension.cs
using LogViewer.Core;
using LogViewer.Core.ViewModels;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace LogViewer.Wpf;

public static class ServicesExtension
{
    public static HostApplicationBuilder AddLogViewer(this HostApplicationBuilder builder)
    {
        builder.Services.AddSingleton<ILogDataStore, Logging.LogDataStore>();
        builder.Services.AddSingleton<LogViewerControlViewModel>();
        builder.Services.AddTransient(service => new LogViewerControl
        {
            DataContext = service.GetRequiredService<LogViewerControlViewModel>()
        });

        return builder;
    }
}

File: VB/Controls/LogViewerVB.Wpf/Extensions/ServicesExtension.vb
Imports System.Runtime.CompilerServices
Imports LogViewerVB.Core
Imports LogViewerVB.Core.ViewModels
Imports Microsoft.Extensions.DependencyInjection
Imports Microsoft.Extensions.Hosting

Public Module ServicesExtension

    <Extension>
    Public Function AddLogViewer(builder As HostApplicationBuilder, Optional config As Action(Of DataStoreLoggerConfiguration) = Nothing) As HostApplicationBuilder

        builder.Services.AddSingleton(Of ILogDataStore, Logging.LogDataStore)
        builder.Services.AddSingleton(Of LogViewerControlViewModel)
        builder.Services.AddTransient(
            Function(service) New LogViewerControl() With
            {
                .DataContext = service.GetRequiredService(Of LogViewerControlViewModel)
            })

        Return builder

    End Function

End Module